src/Databases/DatabasesCommon.cpp

```diff
@@ -166,4 +168,66 @@ StoragePtr DatabaseWithOwnTablesBase::getTableUnlocked(const String & table_name
                     backQuote(database_name), backQuote(table_name));
 }
 
+/// Daisy: starts.
+StorageInMemoryCreateQuery parseCreateQueryFromAST(const ASTPtr & query, const String & database_, const String & table_)
+{
+    return parseCreateQueryFromAST(query.get(), database_, table_);
+}
+
+StorageInMemoryCreateQuery parseCreateQueryFromAST(const IAST * query, const String & database_, const String & table_)
+{
+    StorageInMemoryCreateQuery parsed_create_query;
+    if (query)
+    {
+        ASTPtr query_clone = query->clone();
+        auto * create = query_clone->as<ASTCreateQuery>();
+
+        if (!create)
+        {
+            WriteBufferFromOwnString query_buf;
+            formatAST(*query, query_buf, true);
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Query '{}' is not CREATE query", query_buf.str());
+        }
+
+        create->attach = false;
+        create->database = database_;
+        create->table = table_;
+
+        /// We remove everything that is not needed for ATTACH from the query.
+        assert(!create->temporary);
+        create->as_database.clear();
+        create->as_table.clear();
+        create->if_not_exists = false;
+        create->is_populate = false;
+        create->replace_view = false;
+        create->replace_table = false;
+        create->create_or_replace = false;
+
+        /// For views it is necessary to save the SELECT query itself, for the rest - on the contrary
+        if (!create->isView())
+            create->select = nullptr;
+
+        create->format = nullptr;
+        create->out_file = nullptr;
+
+        /// parse 'query_uuid'
+        parsed_create_query.query_uuid = queryToString(*create);
```

Comment:
i don't understand how query_uuid is parsed by calling `queryToString(ast)` ?

Reply:
the config `show_table_uuid_in_table_create_query_if_not_nil` is dynamic, such as
```
$CLICKHOUSE_CLIENT --show_table_uuid_in_table_create_query_if_not_nil=0 -q "SHOW CREATE DATABASE test_01114_1"
$CLICKHOUSE_CLIENT --show_table_uuid_in_table_create_query_if_not_nil=1 -q "SHOW CREATE DATABASE test_01114_2"
```
so, we must be cache both all